src/js/modules/rpc/server.ts

```diff
@@ -31,10 +31,52 @@ import errors, { DisableResponseCode } from "./errors";
 import { Logger } from "winston";
 import Logging from "../utilities/Logging";
 import requireNative from "./require-native";
+import { isNtpEqual } from "../utilities/domain";
+import { find } from "../utilities/Map";
+
+function groupByTopic(
+  processBatchReplyItems: ProcessBatchReplyItem[]
+): ProcessBatchReplyItem[] {
+  interface Map {
+    key: [ProcessBatchReplyItem["ntp"], ProcessBatchReplyItem["coprocessorId"]];
+    value: ProcessBatchReplyItem;
+  }
+
+  const values = processBatchReplyItems
+    .reduce((prev, result) => {
+      const prevProcessBatch = find(
+        prev,
+        ([ntp, coprocId]) =>
+          isNtpEqual(result.ntp, ntp) && coprocId === result.coprocessorId
+      );
+
+      if (prevProcessBatch === undefined) {
+        return prev.set([result.ntp, result.coprocessorId], result);
+      } else {
+        const [id, prevResult] = prevProcessBatch;
+        if (
+          prevResult.resultRecordBatch === undefined ||
+          result.resultRecordBatch === undefined
+        ) {
+          return prev.set(id, { ...prevResult, resultRecordBatch: undefined });
+        }
+        const newResult: ProcessBatchReplyItem = {
+          ...prevResult,
+          resultRecordBatch: prevResult.resultRecordBatch.concat(
+            result.resultRecordBatch
+          ),
+        };
+        return prev.set(id, newResult);
```

Comment:
Can we just do `prevResult.recordBatch.concat()` and avoid calling `set`?

Reply:
I'm prefer guarantee change the value on Map, change by reference in some cases can generate problems